cmd/jujud/agent/model/manifolds.go

```diff
@@ -472,15 +482,15 @@ func CAASManifolds(config ManifoldsConfig) dependency.Manifolds {
 	modelTag := agentConfig.Model()
 	manifolds := dependency.Manifolds{
 		// The undertaker is currently the only ifNotAlive worker.
-		undertakerName: ifNotUpgrading(ifNotAlive(undertaker.Manifold(undertaker.ManifoldConfig{
+		undertakerName: ifNotAlive(undertaker.Manifold(undertaker.ManifoldConfig{
```

Comment:
We don't want the undertaker to run if the controllers are upgrading

Reply:
this is the model upgrading not the controller.

Reply:
The model upgrading is for the environ only, we don't run the environ upgrader when the model is dying. Otherwise we could never destroy the model if the environ was dying and not upgradable.